openfpga_flow/scripts/run_fpga_flow.py

```diff
@@ -489,6 +489,21 @@ def run_yosys_with_abc():
         "LUT_SIZE": lut_size,
         "OUTPUT_BLIF": args.top_module+"_yosys_out.blif",
     }
+
+    for indx in range(0, len(OpenFPGAArgs), 2):
+        tmpVar = OpenFPGAArgs[indx][2:].upper()
+        ys_params[tmpVar] = OpenFPGAArgs[indx+1]
+    
+    if 'YOSYS_FAMILY' not in ys_params.keys():
+        # define default family as 'qlf_k4n8'
+        ys_params['YOSYS_FAMILY'] = "qlf_k4n8"
+
+    # prefix value of YOSYS_MODE with '-' as an option in yosys script
+    if 'YOSYS_MODE' in ys_params.keys():
+        ys_params['YOSYS_MODE'] = "-" + ys_params['YOSYS_MODE']
+    else:
+        ys_params['YOSYS_MODE'] = ""
+
     yosys_template = args.yosys_tmpl if args.yosys_tmpl else os.path.join(
```

Comment:
This parameter is specific to adder mode, similarly we will have another one for ff tech map. Can we make this variable name specific to adder e.g. YOSYS_ADDER_MODe

Reply:
@tpagarani I think we should avoid adding more specific variables to the script. It will make the python script strongly biased and hard to maintain (think about 100+ variables).
What @ganeshgore proposed is a generic support. Users can define any variable in their yosys script and use the variable in task configuration file. The python script does not check the correctness of the variables. It is users' responsibility to ensure their variables are valid.

Reply:
ok. so what you are saying is that we can have following line in .conf file 
yosys_mode = -no_adder -no_ff 
That will take care of all the flags in this single variable.

Reply:
Yes. It is a solution. Alternatively, you can also consider

- In the yosys script
```
synth_quicklogic -blif ${OUTPUT_BLIF} -family ${YOSYS_FAMILY} -top ${TOP_MODULE} ${YOSYS_ADDER_MODE} ${YOSYS_SHIFT_REGISTER_MODE}
```
 
- In the task configuration file
```
yosys_adder_mode=-no_adder
yosys_shift_register_mode=-no_ff
```

Reply:
@lnsharma  @tpagarani  You can even go generic in configuration file, like
`yosys_args=-family qlf_k4n8  -no_adder`

and in .ys template 
`synth_quicklogic -blif ${OUTPUT_BLIF}-top ${TOP_MODULE}  ${YOSYS_ARGS} `

in that case you  you can drop  this part 
https://github.com/lnis-uofu/OpenFPGA/blob/2b2acae75795c850675496d069eda5cebc6bc706/openfpga_flow/scripts/run_fpga_flow.py#L497-L505

Reply:
I like this one - adding only one variable YOSYS_ARGS and let users choose what they want to specify.